### DISCUSSION TITLE
Input Fields Clearing Unexpectedly with wire:model.live and wire:input

### DISCUSSION THREAD

**Discussion by sobancodes** [livewire/livewire#8992] [1. Help]
I'm experiencing an issue in my Laravel application deployed on Vapor where text in Livewire input fields bound with wire:model.live or wire:input occasionally disappears or clears unexpectedly. During real-time updates, the input fields reset and recent changes are lost. This issue doesn’t occur in my local environment.

**What I’ve Tried:**

Added .debounce.250ms to slow down updates, but this doesn’t fully prevent resets.
This behavior only affects wire:model.live and wire:input; standard wire:model bindings work as expected since they don’t trigger network requests with every keystroke.

**Here’s a video demonstrating the issue for more context:**

https://github.com/user-attachments/assets/83ae99db-bed8-4879-8715-6de34e6f3607

**Environment:**

Laravel (10.48.15)
Livewire (3.5.2)
Vapor

**Goal:** I need these input fields to retain their content on real-time updates without clearing unexpectedly.

Any guidance or insight would be greatly appreciated!

**Edit:** I also tried throttling and added the latency of 1500ms. However, I could still not replicate the problem on local!

**Comment by lrljoe** [comment]
You need to share your actual code for anyone to understand/assist.

That's:
1. Your Component code
2. Any Form component code
3. The blade/view code.

In general - 
Using "Live" on a text field, means that as soon as the "key-up" is fired, it'll fire a backend request.
Now, if your backend isn't lightning fast (read < 50ms) you're going to get a clash between what the backend is returning, and what the user has now updated the frontend to be with their next keypress.

**Comment by nelo-digitalagentur** [comment]
Running into the same behaviour here. Simple Livewire component with an attribute that uses wire:model.live with a input field... typing in the field is "interrupted" by the updated commands leading to loss of the input value when a user continues to type. Seems that Livewire after updating the input field value even selects the whole content on the input field....

It seems that without tweaking the behaviour here one is not supposed to use wire:model.live with a text input field?


https://github.com/user-attachments/assets/40d3b439-d706-41a4-ad4b-828a4d9a5674

**Comment by firer777** [comment]
issue when using nested property like: wire:model.live="post.name", normal when use like wire:model.live="name"

**Reply by lrljoe** [reply]
@sinhnv1991 By nested property, are you talking an array, a Form object or what?

**Reply by firer777** [reply]
Having problems using arrays, I haven't tried it with Form objects, I created an issue a long time ago but haven't seen them solve it, https://github.com/livewire/livewire/discussions/8325.

**Reply by lrljoe** [reply]
> Having problems using arrays, I haven't tried it with Form objects, I created an issue a long time ago but haven't seen them solve it, https://github.com/livewire/livewire/discussions/8325.

That wire box in your linked issue works fine?  The slug in the array updates when updating the name in the array.

Can you provide a wirebox where it isn't working?

**Reply by firer777** [reply]
@lrljoe yes wire box work fine, https://wirebox.app/b/x2180, you type on input have placeholder state.name, when you type at a fast speed you will see the problem of lost letters

**Reply by firer777** [reply]
You will see the problem more clearly when you f12 select the network tab to adjust slow 4G or 3G

**Reply by lrljoe** [reply]
> You will see the problem more clearly when you f12 select the network tab to adjust slow 4G or 3G

Yep, that's what happens if you use live.  Add the debounce with a delay as an example, instead of plain live.

Otherwise you're making network requests every time key up occurs, which can be lossy with slower connections

**Reply by lrljoe** [reply]
https://livewire.laravel.com/docs/wire-model#all-available-modifiers

For options

**Reply by firer777** [reply]
@lrljoe but on input not use nested property is work fine. I also added debounce but it doesn't completely prevent text loss when using nested property

**Reply by lrljoe** [reply]
> @lrljoe but on input not use nested property is work fine. I also added debounce but it doesn't completely prevent text loss when using nested property

Something like
```
wire:model.live.debounce.400ms="nested.property"
```

What does that do in terms of sorting it? It's worth considering disabling the input during the wire:loading phase too.

**Reply by lrljoe** [reply]
Ultimately, live is exactly that, debouncing is just a delay.

If a user is typing in a bound input, after the request to the backend is dispatched, then that will be lost, if you're updating the field they are using. or sent as additional network requests, which will then possibly cause other conflicts.

Disabling the input during loading is the best way to prevent this

**Reply by firer777** [reply]
I even used 1000ms but but it doesn't completely prevent text loss. I think the best solution like some js frameworks have done is to cancel old requests and only send the latest request. For example, if you click page 1,2,3 continuously, it will cancel the pagination request 1,2 and only send request page 3.

**Reply by firer777** [reply]
@lrljoe like this: https://github.com/livewire/livewire/issues/724

**Comment by sobancodes** [comment]
Hi sorry, I don't know how I missed the notifications to this thread but we somehow managed to solve this issue **if your app is deployed on vapor**

According to the their support:

> The issue might be because within the app load balancer in Vapor, AWS is stripping empty headers.

as a result the X-Livewire request header was being stripped away. So You can set the X-Livewire header in `app.js`. Or you can also refer to this comment https://github.com/livewire/livewire/discussions/8333#discussioncomment-10345370 which helped me solve it. Thank you
